h5py/h5ds.pyx

```diff
@@ -37,28 +37,54 @@ def is_scale(DatasetID dset not None):
 @with_phil
 def attach_scale(DatasetID dset not None, DatasetID dscale not None, unsigned
                  int idx):
+    """(DatasetID dset, DatasetID dscale, UINT idx)
+
+    Attach Dimension Scale dscale to Dimension idx of Dataset dset.
+    """
     H5DSattach_scale(dset.id, dscale.id, idx)
 
 @with_phil
 def is_attached(DatasetID dset not None, DatasetID dscale not None,
                 unsigned int idx):
+    """(DatasetID dset, DatasetID dscale, UINT idx)
```

Comment:
```suggestion
    """(DatasetID dset, DatasetID dscale, UINT idx) => BOOL
```

Reply:
> When successful, returns a positive value for TRUE or 0 (zero) for FALSE. Otherwise returns a negative value.

I realized that the returned value are not just Boolean value(True/False), so I removed `=> BOOL` after I added it in the first.

Reply:
The HDF5 C function returns integers, but the h5py wrapper converts them into boolean values. The `<bint>` in the Cython code is casting to a *boolean integer*, a special Cython type to bridge between Python booleans and C. When it's returned as a Python object, it's automatically converted to a Python bool.

We can do this partly because Python has a separate mechanism for exceptions, whereas the C API needs to use negative numbers to indicate errors.